common/protos/common/asset_locks.proto

```diff
@@ -19,18 +19,25 @@ message AssetClaim {
   bytes claimInfo = 2;
 }
 
+enum HashMechanism {
+  SHA256 = 0;
+  SHA512 = 1;
```

Comment:
If we are including this as an option here, we should offer SHA512-based locking and claiming in the Interop Modules too.

Reply:
Done @VRamakrishna .